src/ts/components/fullscreentogglebutton.ts

```diff
@@ -28,17 +28,34 @@ export class FullscreenToggleButton extends ToggleButton<ToggleButtonConfig> {
       }
     };
 
+    let fullscreenAvailableHandler = () => {
+      if (player.isViewModeAvailable(player.exports.ViewMode.Fullscreen)) {
+        this.show();
+      } else {
+        this.hide();
+      }
+    };
+
+    uimanager.getConfig().events.onUpdated.subscribe(fullscreenAvailableHandler);
```

Comment:
I suppose that the `onUpdated` event of the `UiManager` signals, that the general config of the UI manager has changed? In that case, `uiConfigUpdatedHandler` may be an even more expressive name for the `fullscreenAvailableHandler`.

Reply:
Most of the time a component just needs to update one particular state when the `UIManager` signals an update. Therefore we pass the callback directly. If there are multiple callbacks which need to be called `onUpdate` I agree with you.